pennylane/tape/qscript.py

```diff
@@ -835,11 +834,6 @@ def copy(self, copy_operations=False):
         )
         new_qscript._graph = None if copy_operations else self._graph
         new_qscript._specs = None
-        new_qscript.wires = copy.copy(self.wires)
-        new_qscript.num_wires = self.num_wires
-        new_qscript._update_par_info()
-        new_qscript._obs_sharing_wires = self._obs_sharing_wires
-        new_qscript._obs_sharing_wires_id = self._obs_sharing_wires_id
```

Comment:
I think these need to stay since these are still private attributes. When copying the tape, the new tape should already have the attributes initialized.

Reply:
But these would be calculated when needed right? Similarly as to why we don't initialize them in the constructor of the QuantumScript. Is this correct?

Reply:
@mudit2812 what do you think?

Reply:
Yes, that's correct. When we copy a tape, since all ops and measurements are the same, these will be correct if they're computed lazily for the copied tape as well. However, copying generally includes _all_ attributes. The final result will be the same regardless of the way we go about this, so it's not really a matter of correctness, but just consistency. 

@albi3ro what are your thoughts on this?

Reply:
Additionally, while users should not be accessing the private attribute, they could, and not having the same value for it as the original tape could impact their workflow.

Reply:
If we went through the effort of calculating something like `wires`, we can keep from having to calculate it again.  Performance shortcut that prevents us from having to do repeat work.

But if we didn't go through the effort of calculating wires and don't already have that value immediately available, we shouldn't have to calculate it here.

Looks like we might be able to do such a check using a syntax like:
```
if "wires" in self.__dict__:
    new_qscript.wires == self.wires
```

But either option (remove or keep) is fine with me at the end of the day.